lib/wab/impl/model.rb

```diff
@@ -101,9 +100,8 @@ def select(format, rid, where, filter)
             }
           else
             @map.each { |ref,obj|
-              if where.eval(obj) && (filter.nil? || filter.eval(obj))
-                matches << extract_matches(format, ref, rid, obj)
-              end
+              next unless where.eval(obj) && (filter.nil? || filter.eval(obj))
+              matches << extract_matches(format, ref, rid, obj)
```

Comment:
Really, how does this improve the code? To me an If statement which identifies a single logical step is more clear that breaking into two separate steps.